community/server/src/main/java/org/neo4j/server/rest/transactional/Neo4jJsonCodec.java

```diff
@@ -71,6 +78,25 @@ else if ( value instanceof Map )
         {
             writeMap(out, (Map) value );
         }
+        else if( value instanceof Geometry )
+        {
+            Geometry geom = (Geometry) value;
+            Object coordinates = (geom instanceof Point) ? ((Point) geom).getCoordinate() : geom.getCoordinates();
```

Comment:
I did have it like that at first but the code was much more verbose with 90% duplication, so I simplified and compacted it to this much shorter version. The only difference between point and all other possible types is that in GeoJSON point is an array of doubles, and others all have arrays of arrays of doubles. A bit inconsistent, but that seems to be the design. By writing the code like this I focus on that difference, I believe, which I think is good. When I had two bigger sections, one for Point and one for Geometry, it was harder to see what the actual differences were.